lib/tbot/config/config.go

```diff
@@ -159,6 +164,37 @@ type OnboardingConfig struct {
 	JoinMethod types.JoinMethod `yaml:"join_method"`
 }
 
+// HasTokenValue gives the ability to check if there has been a token value stored
+// in the config
+func (conf *OnboardingConfig) HasTokenValue() bool {
+	return conf.TokenValue != ""
+}
+
+// StoreToken stores the value for --token or auth_token in the config
+//
+// In the case of the token value pointing to a file, this allows us to
+// fetch the value of the token when it's needed (when connecting for the first time)
+// instead of trying to read the file every time that teleport is launched.
+// This means we can allow temporary token files that are removed after teleport has
+// successfully connected the first time.
+func (conf *OnboardingConfig) StoreToken(token string) {
+	conf.TokenValue = token
```

Comment:
Do you need a setter if it's an exported field?

Reply:
No, I was just keeping a similar interface between this and my other PR